packages/editor/src/components/browser/role/AddRolePopover.tsx

```diff
@@ -70,28 +70,31 @@ export const AddRolePopover = ({
         />
       </PopoverTrigger>
       <PopoverContent sideOffset={12} collisionPadding={5} style={{ zIndex: '10000' }}>
-        <Flex direction='column' gap={2} alignItems='center'>
-          <BasicField label='New role name' style={{ width: '100%' }}>
-            <Input value={newRoleName} onChange={e => setNewRoleName(e.target.value)} />
-          </BasicField>
-          {newNameExists(table, newRoleName) && <Message variant='error' message='A role with that name already exists' />}
-          <Button
-            icon={IvyIcons.Plus}
-            onClick={() => {
-              addRole.mutate({
-                context,
-                newRole: { identifier: newRoleName, parent: value }
-              });
-              setAddedRoleName(newRoleName);
-            }}
-            aria-label='Add new Role'
-            title='Add new Role'
-            disabled={!newNameIsValid(table, newRoleName)}
-            style={{ width: '100%' }}
-          >
-            Add Role to {value}
-          </Button>
-        </Flex>
+        <form onSubmit={event => event.preventDefault()}>
+          <Flex direction='column' gap={2} alignItems='center'>
+            <BasicField label='New role name' style={{ width: '100%' }}>
+              <Input value={newRoleName} onChange={e => setNewRoleName(e.target.value)} />
+            </BasicField>
+            {newNameExists(table, newRoleName) && <Message variant='error' message='A role with that name already exists' />}
+            <Button
+              icon={IvyIcons.Plus}
+              onClick={() => {
+                addRole.mutate({
+                  context,
+                  newRole: { identifier: newRoleName, parent: value }
+                });
+                setAddedRoleName(newRoleName);
+              }}
+              aria-label='Add new Role'
+              title='Add new Role'
+              disabled={!newNameIsValid(table, newRoleName)}
+              style={{ width: '100%' }}
+              variant='primary'
```

Comment:
if you use a from with onSubmit then I think you should add the `type='submit'` to the button.